Only unset loop when with_loop

ruudk opened the pull request:
Even though PHP does not complain, PHPStan does.
> Cannot unset offset 'loop' on array{ ... the context values ... }

Since this is compiled code, we can easily produce a bit more valid code in the eyes of PHPStan.

fabpot commented on src/Node/ForNode.php:
```diff
@@ -101,7 +101,11 @@ public function compile(Compiler $compiler): void
+        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']);'."\n");
+
+        if ($this->getAttribute('with_loop')) {
+            $compiler->write('unset($context[\'loop\']);'."\n");
```
Can we use a single `unset()` call in this case?

ruudk replied:
Sure, done!

stof commented on src/Node/ForNode.php:
```diff
@@ -101,7 +101,11 @@ public function compile(Compiler $compiler): void
-        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\'], $context[\'loop\']);'."\n");
+        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']');
+        if ($this->getAttribute('with_loop')) {
+            $compiler->write(', $context[\'loop\']');
```
this needs to use `->raw` so that it does not add indentation again (as it is not meant to write new lines but to continue the content of the current line).
Same below for the end of the line

ruudk replied:
Good to know, changed!

ruudk committed:
Only unset loop when with_loop

Even though PHP does not complain, PHPStan does.

Since this is compiled code, we can easily produce a bit more valid code in the eyes of PHPStan.

fabpot commented:
Thank you @ruudk.
